projectgenerator/utils/qgis_utils.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+    begin                :    05/03/18
+    git sha              :    :%H$
+    copyright            :    (C) 2018 by Germán Carrillo (BSF-Swissphoto)
+    email                :    gcarrillo@linuxmail.org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+from qgis.core import QgsLayerTreeNode, QgsProject, QgsWkbTypes
+
+layer_order = [QgsWkbTypes.PointGeometry,
+               QgsWkbTypes.LineGeometry,
+               QgsWkbTypes.PolygonGeometry,
+               QgsWkbTypes.UnknownGeometry]
+
+def get_first_index_for_geometry_type(geometry_type, group=QgsProject.instance().layerTreeRoot()):
```

Comment:
The layer tree should be passed through to here to not depend on QgsProject being a singleton

Reply:
Where could we get the layer tree from?

Reply:
Up there you have a reference to qgis_project https://github.com/opengisch/projectgenerator/pull/147/files#diff-5e65be2a57725b9de36675393dc84b8dR70

Reply:
Done!